### PR TITLE
Refactor the index file format

### PR DESCRIPTION
**Pull request by longjiquan**
Signed-off-by: dragondriver <jiquan.long@zilliz.com>
issue: #8513

### CODE REVIEW

**Comment by DragonDriver** [comment]
/hold
TODO:
1, ut of this feature;
2, integrates the new file format to indexnode and querynode;
3, test the ci cases;

File: internal/storage/binlog_writer.go [hunk 1]
```diff
@@ -269,3 +286,36 @@ func NewDDLBinlogWriter(dataType schemapb.DataType, collectionID int64) *DDLBinl
...
+func NewIndexFileBinlogWriter(
+	indexBuildID UniqueID,
+	version int64,
+	collectionID UniqueID,
+	partitionID UniqueID,
+	segmentID UniqueID,
+	fieldID UniqueID,
+	indexName string,
+	indexID UniqueID,
+	key string,
```
[the remark this answers is not in the document]
Comment (xiaofan-luan) [reply internal/storage/binlog_writer.go:299]: It would be better if we add a parameter

[the remark this answers is not in the document]
Comment (DragonDriver) [reply internal/storage/binlog_writer.go:299]: The key of index file is important in knowhere.
Different index type may have different key.
So encode the key to binlog makes it a totally self-analysis file.

**Pull request merged**